include/mbgl/actor/actor.hpp

```diff
@@ -8,9 +8,108 @@
 #include <memory>
 #include <future>
 #include <type_traits>
+#include <cassert>
 
 namespace mbgl {
 
+template <class Object>
+class EstablishedActor;
+
+template <class Object>
+class Actor;
+
+template <class Object>
+class AspiringActor {
+public:
+    AspiringActor() : mailbox(std::make_shared<Mailbox>()) {
+        assert(!mailbox->isOpen());
+    }
+    
+    AspiringActor(const AspiringActor&) = delete;
+    
+    template <class... Args>
+    static std::tuple<std::decay_t<Args>...> captureArguments(Args&&... args) {
```

Comment:
This doesn't seem like it really belongs in `AspiringActor`. Since the call site needs to use `std::forward<Args>(args)...` anyway, I would just inline `std::make_tuple(...)` as well.